tetris.py

```diff
@@ -125,23 +125,55 @@ def new_block(self):
             self.current_block_shape = self.upcoming_block_shape
         self.current_block_center = (int(game_width / 2), -2)
         self.upcoming_block_shape = random.choice(shape_names)
+        self.index = 0
 
     def get_current_block(self):
         (x, y) = self.current_block_center
         if self.current_block_shape == "I":
-            return [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)]
+            I = [
+                [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)],
+                [(x, y - 2), (x, y - 1), (x, y), (x, y + 1)],
+            ]
+            return I[self.index % len(I)]
         if self.current_block_shape == "L":
-            return [(x - 1, y - 1), (x - 1, y), (x, y), (x + 1, y)]
+            L = [
+                [(x - 1, y), (x, y), (x + 1, y), (x + 1, y - 1)],
+                [(x + 1, y), (x, y), (x, y - 1), (x, y - 2)],
+                [(x, y), (x, y - 1), (x + 1, y - 1), (x + 2, y - 1)],
+                [(x, y - 1), (x + 1, y - 1), (x + 1, y), (x + 1, y + 1)],
```

Comment:
This code is difficult to read. The coordinates don't follow any kind of obvious pattern. For example, the first coordinates of each list are `(x-1, y), (x+1, y), (x,y), (x, y-1)`. I have no idea whether that's what they should be.

If you would simply rotate around `(x, y)`, going 90 degrees clockwise at a time, you would get `(x-1, y), (x, y-1), (x+1,y), (x, y+1)`. This pattern is easy to pick up when reading to code, and so it's possible to properly review this code and check that the list is correct.